### PR TITLE
feat: add natspec-smells

### PR DESCRIPTION
**Pull request by dristpunk**
Closes BES-236

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
   "devDependencies": {
     "@commitlint/cli": "17.0.3",
     "@commitlint/config-conventional": "17.0.3",
+    "@defi-wonderland/natspec-smells": "^1.0.2",
```
Comment (gas1cent) [inline package.json:40]:
```suggestion
    "@defi-wonderland/natspec-smells": "1.0.2",
```

File: natspec-smells.config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
+ * List of supported options: https://github.com/defi-wonderland/natspec-smells?tab=readme-ov-file#options
```
Comment (0xJabberwock) [inline natspec-smells.config.js:2]: Observation: `constructorNatspec` option is by default set to false. 🤔

Comment (gas1cent) [reply natspec-smells.config.js:2]: @dristpunk let's see if most projects you have been testing the tool with have constructor natspec. I argue it's not that popular and hence should be `false` even in our own projects.

**Review by 0xJabberwock — APPROVED** [review]

**Review by gas1cent — APPROVED** [review]

File: natspec-smells.config.js (hunk 2 above)
Comment (dristpunk) [reply natspec-smells.config.js:2]: we decided to leave it as it is for now

**Review by 0xRaccoon — APPROVED** [review]

**Pull request merged**